app/code/Magento/Inventory/Model/IsProductSalable.php

```diff
@@ -86,55 +58,7 @@ public function execute(string $sku, int $stockId): bool
 
         $isSalable = (bool)$stockItemData['is_salable'];
         $qtyWithReservation = $stockItemData['quantity'] + $this->getReservationsQuantity->execute($sku, $stockId);
-        $globalMinQty = $this->configuration->getMinQty();
-        $legacyStockItem = $this->getLegacyStockItem($sku);
-        if (null === $legacyStockItem) {
-            return false;
-        }
-
-        if ($this->isManageStock($legacyStockItem)) {
-            if (($legacyStockItem->getUseConfigMinQty() == 1 && $qtyWithReservation <= $globalMinQty)
-                || ($legacyStockItem->getUseConfigMinQty() == 0 && $qtyWithReservation <= $legacyStockItem->getMinQty())
-            ) {
-                $isSalable = false;
-            }
-        }
-
-        return $isSalable;
-    }
-
-    /**
-     * @param LegacyStockItem $legacyStockItem
-     *
-     * @return bool
-     */
-    private function isManageStock(LegacyStockItem $legacyStockItem): bool
-    {
-        $globalManageStock = $this->configuration->getManageStock();
-        $manageStock = false;
-        if (($legacyStockItem->getUseConfigManageStock() == 1 && $globalManageStock == 1)
-            || ($legacyStockItem->getUseConfigManageStock() == 0 && $legacyStockItem->getManageStock() == 1)
-        ) {
-            $manageStock = true;
-        }
-
-        return $manageStock;
-    }
-
-    /**
-     * @param string $sku
-     *
-     * @return LegacyStockItem|null
-     */
-    private function getLegacyStockItem(string $sku)
-    {
-        $productIds = $this->productResource->getProductsIdsBySkus([$sku]);
-        $searchCriteria = $this->stockItemCriteriaFactory->create();
-        $searchCriteria->addFilter(StockItemInterface::PRODUCT_ID, StockItemInterface::PRODUCT_ID, $productIds[$sku]);
-
-        $legacyStockItem = $this->legacyStockItemRepository->getList($searchCriteria);
-        $items = $legacyStockItem->getItems();
 
-        return count($items) ? reset($items) : null;
+        return $this->stockConfiguration->execute($sku, $stockId, $qtyWithReservation, $isSalable);
```

Comment:
What we want to do here - is to apply all the validations based on stock configuration. Like minimal quantity, is mange stock etc.

Reading code - it's not clear what is going on here
`stockConfiguration->execute($sku, $stockId, $qtyWithReservation, $isSalable)`
And what we want to achieve.

We are applying validations and additional checks based on inventory configuration. 
Thus, each validator will use Configuration API to extract particular value. 

But here you will never guess that untill you start debugging the implementation. 
High-level code should be self-descriptive - you should read it as straightforward as a business logic.